graph-partitioning/graph_partitioning.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2019 D-Wave Systems, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# ------ Import necessary packages ----
+import networkx as nx 
+from collections import defaultdict
+from itertools import combinations
+from dwave.system.samplers import DWaveSampler
+from dwave.system.composites import EmbeddingComposite
+
+# ------- Set tunable parameters -------
+num_reads = 1000
+gamma = 156
```

Comment:
This is a surprisingly precise number

Reply:
40 node graphs can have at most 780 edges, but we're only choosing 20% of them for an estimated number of 156 edges in our graph.